larpandoracontent/LArTrackShowerId/TrackShowerIdFeatureTool.cc

```diff
@@ -469,9 +470,16 @@ void ThreeDVertexDistanceFeatureTool::Run(LArMvaHelper::MvaFeatureVector &featur
         try
         {
             vertexDistance = (pVertexList->front()->GetPosition() - LArPfoHelper::GetVertex(pInputPfo)->GetPosition()).GetMagnitude();
+            error = false;
         }
         catch (const StatusCodeException &) {}
     }
+    if (error == true)
+    {
+        CaloHitList threeDCaloHitList;
+        LArPfoHelper::GetCaloHits(pInputPfo, TPC_3D, threeDCaloHitList);
+        vertexDistance = (pVertexList->front()->GetPosition() - (threeDCaloHitList.front())->GetPositionVector()).GetMagnitude(); // if n3dHits == 1, can't calculate vertex postion of input pfos hence default to 10.0 based on this variable's distribution to make it unbiased towards tracks or showers
```

Comment:
It's not obvious to me how this calculation corresponds to the comment, can you comment further on this?

Reply:
This basically says that if you have a PFO with 1 3D hit (which is suprisingly possible), for some reason, it fails to execute line 472: `LArPfoHelper::GetVertex(pInputPfo)->GetPosition()).GetMagnitude()`.  This ends up throwing an error. To bypass this, I just asked for the position of the 1 3d hit instead and then set vertexDistance to be the magnitude of the difference between the interaction vertex and the hit's position.

Reply:
Thanks for the confirmation; though I think that makes the comment alongside line 481 slightly outdated - presumably the result of this calculation is not always 10?

Reply:
This is correct. My bad. I will fix it soon.